actix-web-grants/src/guards.rs

```diff
@@ -29,22 +29,54 @@ use std::hash::Hash;
 ///    Ok(HashSet::from(["ROLE_ADMIN".to_string()]))
 /// }
 /// ```
-pub struct AuthorityGuard<Type> {
-    allow_authority: Type,
+
+pub struct AuthorityGuard<T> {
+    allow_authority: Type<T>,
+}
+
+pub enum Type<T> {
+    Single(T),
+    Any(Vec<T>),
+    All(Vec<T>),
 }
 
-impl<Type: Eq + Hash + 'static> AuthorityGuard<Type> {
-    pub fn new(allow_authority: Type) -> AuthorityGuard<Type> {
```

Comment:
Removing `new` is breaking change.
Let’s probably keep it as an alias (perhaps, deprecated)